usr/sbin/p11sak/p11sak.c

```diff
@@ -1200,11 +1268,119 @@ static const struct p11sak_arg p11sak_copy_key_args[] = {
+                     "letters.", },
+    { .short_opt = 'i', .long_opt = "id", .required = false,
+      .arg =  { .type = ARG_TYPE_STRING, .required = true,
+                .value.string = &opt_id, .name = "ID", },
+      .description = "The ID of the key to be imported. Specify a hex string "
+                     "(not prefixed with 0x) of any number of bytes.", },
+    { .short_opt = 'F', .long_opt = "file", .required = true,
+      .arg =  { .type = ARG_TYPE_STRING, .required = true,
+                .value.string = &opt_file, .name = "FILENAME", },
+      .description = "The file name of the file that contains the key to be "
+                     "imported. For symmetric keys, this is a binary file "
+                     "containing the key material in clear. For asymmetric "
+                     "keys, this is an OpenSSL PEM file containing a "
+                     "public or private key. PEM files can optionally be "
+                     "password protected. Specify the PEM password with the "
+                     "'-P'/'--pem-password' option or environment variable "
+                     "P11SAK_PEM_PASSWORD. If the PEM file is password "
+                     "protected, but no PEM password is specified, you will be "
+                     "prompted for the PEM password.", },
+    { .short_opt = 'P', .long_opt = "pem-password", .required = false,
+      .arg =  { .type = ARG_TYPE_STRING, .required = true,
+                .value.string = &opt_pem_password, .name = "PASSWORD", },
+      .description = "The password of the PEM file specified with the "
+                     "'-F'/'--file' option. If the PEM file is password "
+                     "protected, but this option is not specified, nor "
+                     "environment variable P11SAK_PEM_PASSWORD is set, you "
+                     "will be prompted for the PEM password.", },
+    { .short_opt = 0, .long_opt = "force-pem-pwd-prompt", .required = false,
+      .long_opt_val = OPT_FORCE_PEM_PWD_PROMPT,
+      .arg = { .type = ARG_TYPE_PLAIN, .required = false,
+               .value.plain = &opt_force_pem_pwd_prompt, },
+      .description = "Enforce PEM password prompt, even if environment "
+                     "variable P11SAK_PEM_PASSWORD is set, or the "
+                     "'-P'/'--pem-password' option is specified.", },
+    { .short_opt = 'o', .long_opt = "opaque", .required = false,
+      .arg = { .type = ARG_TYPE_PLAIN, .required = false,
+               .value.plain = &opt_opaque, },
+      .description = "The key material in the file specified with the "
+                     "'-F'/'--file' option is an opaque secure key blob. "
+                     "Not all tokens support this.", },
+    { .short_opt = 0, .long_opt = NULL, },
+};
+
+static const struct p11sak_enum_value p11sak_import_asym_types[] = {
+    { .value = "public", .args = NULL, .private = { .num = false }, },
+    { .value = "private", .args = NULL, .private = { .num = true }, },
```

Comment:
Here is something wrong ... 1x public 3x private ?

Reply:
Ahm no.... the "public" and "private" used for `.value` is the enumeration text that the user specifies. 
The `.private` is a way to attach 'private' data to that enumeration value. Here 'private' in terms of it belongs to that enum value, not in terms of private vs public.